cpp/arcticdb/column_store/column.hpp

```diff
@@ -800,10 +815,26 @@ class Column {
         return orig_type_.value();
     }
 
+    void set_secondary_type(TypeDescriptor type) {
+        secondary_type_ = std::move(type);
+    }
+
+    bool has_secondary_type() const {
+        return static_cast<bool>(secondary_type_);
+    }
+
+    const TypeDescriptor& secondary_type() const {
+        return *secondary_type_;
```

Comment:
This will error out if `secondary_type_` is `std::nullopt`. `util::check` first?